tests/behaviour/steps/cli.py

```diff
@@ -87,15 +87,30 @@ def step_impl(context):
 
 def run_command_in_temp_dir(context, command: str) -> Tuple[int, str]:
     tmp_dir_path = get_context_temp_dir_path(context)
-    process = subprocess.Popen(
-        command,
-        shell=True,
-        cwd=tmp_dir_path,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-    )
-    status_code = process.wait()
-    response = process.stdout.read().decode("utf-8") + process.stderr.read().decode(
-        "utf-8"
-    )
+
+    # Use temp files not a PIPE, a PIPE has a tiny buffer than
+    # can deadlock or result in eroneous resource exhaustion behaviour 
+    # where encountering some of our larger outputs (jsonSchemaErrors result
+    # in large writes to stdout)
+    Path(tmp_dir_path / "buffer").mkdir()
+    stdout_path = Path(tmp_dir_path / "buffer" / "stdout")
+    stderr_path = Path(tmp_dir_path / "buffer" / "stderr")
+
+    with open(stdout_path, "w") as stdout_file:
```

Comment:
You can do multiple `with` bindings on the same line (so you don't have to indent twice) 

e.g. 

```python
with A() as a, B() as b:
    suite
```

https://docs.python.org/3.1/reference/compound_stmts.html#with

Reply:
I had no idea that was a thing, cool :smile: